cpp/src/io/parquet/writer_impl.cu

```diff
@@ -1171,23 +1253,47 @@ void writer::impl::write(table_view const& table)
   // Build chunk dictionaries and count pages
   if (num_chunks != 0) { init_page_sizes(chunks, col_desc, num_columns); }
 
+  // Get the maximum page size across all chunks
+  size_type max_page_uncomp_data_size =
+    std::accumulate(chunks.host_view().flat_view().begin(),
+                    chunks.host_view().flat_view().end(),
+                    0,
+                    [](uint32_t max_page_size, gpu::EncColumnChunk const& chunk) {
+                      return std::max(max_page_size, chunk.max_page_data_size);
```

Comment:
I think you can use `max_element` here

Reply:
The replacement seems worse
```c++
  auto max_page_data_size_accessor = [&](gpu::EncColumnChunk const* chunk) {
    return thrust::make_transform_iterator(
      chunk, [](gpu::EncColumnChunk const& chunk) { return chunk.max_page_data_size; });
  };
  auto max_page_uncomp_data_size_it =
    std::max_element(max_page_data_size_accessor(chunks.host_view().flat_view().begin()),
                     max_page_data_size_accessor(chunks.host_view().flat_view().end()));
  size_t max_page_uncomp_data_size =
    (max_page_uncomp_data_size_it !=
     max_page_data_size_accessor(chunks.host_view().flat_view().end()))
      ? *max_page_uncomp_data_size_it
      : 0;
```

Reply:
does this work?
```
  auto const max_page_uncomp_data_size =
    std::max_element(chunks.host_view().flat_view().begin(),
                     chunks.host_view().flat_view().end(),
                     [](auto& l, auto& r) { return l.max_page_data_size < r.max_page_data_size; })
      ->max_page_data_size;
```
I don't understand why you need to check the result. Is the input empty in some cases?

Reply:
>Is the input empty in some cases?

Yes. It broke in the empty output test

Reply:
Can be worked around with a ternary, but the original code covers the empty case implicitly so it's not clearly better to use `max_element` here. Darn :)